### ISSUE TITLE
[Misc]: cannot run docker image 0.5.0 with cuda version == 12.1

### ISSUE THREAD

**Issue by kratorado** [vllm-project/vllm#5635]
### Anything you want to discuss about vllm.

I have ubuntu-22.04, cuda==12.1, nvidia driver == 530, gpu==V100

when running newer version of vllm (after 0.4.3) , it does not work. According to the doc, vllm is built with cuda 12.1, so I think it should not require cuda>=12.4.

Error log
```
docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: Auto-detected mode as 'legacy'
nvidia-container-cli: requirement error: unsatisfied condition: cuda>=12.4, please update your driver to a newer version, or use an earlier cuda container: unknown.
```

**Comment by mohit-zangoh** [comment]
+1

**Comment by kratorado** [comment]
found the related docker commands 
https://hub.docker.com/layers/vllm/vllm-openai/v0.4.3/images/sha256-d55c38328bfda7b049dd6c3695d60fff8d88a63b19db058791879dc6bf3188af?context=explore

what new features require cuda>=12.4 ?

**Comment by youkaichao** [comment]
we use 12.4 because the old versions of base container will be deleted soon, according to nv policy https://gitlab.com/nvidia/container-images/cuda/-/blob/master/doc/support-policy.md .

**Comment by pseudotensor** [comment]
It seems even v0.4.3 is now built with cuda 12.4 required.

And v0.4.2 fails with:
```
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.10/dist-packages/vllm/entrypoints/openai/api_server.py", line 26, in <module>
    from vllm.entrypoints.openai.serving_chat import OpenAIServingChat
  File "/usr/local/lib/python3.10/dist-packages/vllm/entrypoints/openai/serving_chat.py", line 20, in <module>
    from vllm.model_executor.guided_decoding import (
  File "/usr/local/lib/python3.10/dist-packages/vllm/model_executor/guided_decoding/__init__.py", line 5, in <module>
    from vllm.model_executor.guided_decoding.lm_format_enforcer_decoding import (
  File "/usr/local/lib/python3.10/dist-packages/vllm/model_executor/guided_decoding/lm_format_enforcer_decoding.py", line 15, in <module>
    from vllm.model_executor.guided_decoding.outlines_decoding import (
  File "/usr/local/lib/python3.10/dist-packages/vllm/model_executor/guided_decoding/outlines_decoding.py", line 15, in <module>
    from vllm.model_executor.guided_decoding.outlines_logits_processors import (
  File "/usr/local/lib/python3.10/dist-packages/vllm/model_executor/guided_decoding/outlines_logits_processors.py", line 24, in <module>
    from outlines.fsm.fsm import CFGFSM, FSM, RegexFSM
  File "/usr/local/lib/python3.10/dist-packages/outlines/__init__.py", line 2, in <module>
    import outlines.generate
  File "/usr/local/lib/python3.10/dist-packages/outlines/generate/__init__.py", line 1, in <module>
    from .api import SequenceGenerator
  File "/usr/local/lib/python3.10/dist-packages/outlines/generate/api.py", line 5, in <module>
    from outlines.fsm.fsm import FSMState
  File "/usr/local/lib/python3.10/dist-packages/outlines/fsm/fsm.py", line 9, in <module>
    from outlines.fsm.regex import create_fsm_index_tokenizer, make_deterministic_fsm
  File "/usr/local/lib/python3.10/dist-packages/outlines/fsm/regex.py", line 96, in <module>
    def create_fsm_info(
  File "/usr/local/lib/python3.10/dist-packages/numba/core/decorators.py", line 229, in wrapper
    disp.enable_caching()
  File "/usr/local/lib/python3.10/dist-packages/numba/core/dispatcher.py", line 856, in enable_caching
    self._cache = FunctionCache(self.py_func)
  File "/usr/local/lib/python3.10/dist-packages/numba/core/caching.py", line 601, in __init__
    self._impl = self._impl_class(py_func)
  File "/usr/local/lib/python3.10/dist-packages/numba/core/caching.py", line 337, in __init__
    raise RuntimeError("cannot cache function %r: no locator available "
RuntimeError: cannot cache function 'create_fsm_info': no locator available for file '/usr/local/lib/python3.10/dist-packages/outlines/fsm/regex.py'

```

**Comment by kratorado** [comment]
> It seems even v0.4.3 is now built with cuda 12.4 required.

Yes. But the document didn't update with that. https://docs.vllm.ai/en/v0.4.3/getting_started/installation.html

**Comment by youkaichao** [comment]
hopefully this can be fixed when i fix https://github.com/vllm-project/vllm/pull/5646

**Comment by kratorado** [comment]
> hopefully this can be fixed when i fix #5646

great!
